tests/chainerx_tests/unit_tests/routines_tests/test_indexing.py

```diff
@@ -368,3 +390,25 @@ def test_where_scalar_scalar(xp, cond_shape, cond_dtype, in_types, out_dtype):
     y = y_type(2)
     out = xp.where(cond, x, y)
     return dtype_utils.cast_if_numpy_array(xp, out, out_dtype)
+
+
+# TODO(aksub99): Add cuda tests after Cumsum cuda kernel is implemented
+@op_utils.op_test(['native:0'])
+@chainer.testing.parameterize_pytest('input', _nonzero_params)
+class TestNonzero(op_utils.NumpyOpTest):
+
+    check_numpy_strides_compliance = False
+    skip_backward_test = True
+    skip_double_backward_test = True
+
+    def setup(self, dtype):
+        a_np = self.input
+        self.a_np = a_np
+
+    def generate_inputs(self):
+        return self.a_np,
+
+    def forward_xp(self, inputs, xp):
+        a, = inputs
+        b = xp.nonzero(a)
```

Comment:
`xp.nonzero` should return a tuple, not list.

Reply:
Yes. `chainerx.nonzero` is returning a list instead of a tuple. How can I change this?

Reply:
I will modify `routines.cc` accordingly.

Reply:
I have modified the implementation and it does return a `tuple`. But I still get the error:
```
    variable.Variable._init_unchecked(
        y, requires_grad=y.is_backprop_required(),
        is_chainerx_array=True)
>   for y in outputs])
E   AttributeError: 'list' object has no attribute 'is_backprop_required'
```

Reply:
I have modified `routines.cc` similar to `split` to try and solve the problem. I am now getting the error:
```
TypeError: nonzero(): incompatible function arguments. The following argument types are supported:
E           1. (a: chainerx::Array) -> List[chainerx::Array]
E       
E       Invoked with: array([[1., 1., 1.],
E              [1., 1., 1.]], shape=(2, 3), dtype=float64, device='native:0', backprop_ids=['<default>'])
```
@asi1024 Do you have any idea how this can be solved?

Reply:
How about trying `py::cast`?